src/main/java/com/authlete/common/api/AuthleteApiBasicImpl.java

```diff
@@ -0,0 +1,1090 @@
+                String responseBody = extractErrorData(ctx);
+                if (responseBody != null && responseClass != null)
+                {
+                    try
+                    {
+                        return Utils.fromJson(responseBody, responseClass);
+                    }
+                    catch (Exception e)
+                    {
+                        // If parsing fails, return default response instead of throwing
+                        return createDefaultErrorResponse(responseClass, status);
+                    }
+                }
+                // No response body or no response class, return default
+                return createDefaultErrorResponse(responseClass, status);
+                
+            case THROW_EXCEPTION:
+            default:
+                // Default behavior: throw exception with context
+                String statusMsg = extractStatusMessage(ctx.connection());
+                Map<String, List<String>> headers = ctx.connection().getHeaderFields();
+                String body = extractErrorData(ctx);
+                throw new AuthleteApiException("HTTP " + status + " " + statusMsg,
+                        status, statusMsg, body, headers);
+        }
+    }
+
+
+    /**
+     * Create a default success response for the given response class.
+     */
+    @SuppressWarnings("unchecked")
+    private <TResponse> TResponse createDefaultSuccessResponse(Class<TResponse> responseClass)
+    {
+        if (responseClass == null)
+        {
+            return null;
+        }
+        
+        // Special handling for ApiResponse
+        if (com.authlete.common.dto.ApiResponse.class.isAssignableFrom(responseClass))
+        {
+            com.authlete.common.dto.ApiResponse response = new com.authlete.common.dto.ApiResponse();
+            response.setResultCode("A999999"); // Generic success code
+            response.setResultMessage("Resource not found (already deleted)");
+            return (TResponse) response;
+        }
+        
+        // For other types, return null
+        return null;
+    }
+
+
+    /**
+     * Create a default error response for the given response class and status code.
+     */
+    @SuppressWarnings("unchecked")
+    private <TResponse> TResponse createDefaultErrorResponse(Class<TResponse> responseClass, int statusCode)
+    {
+        if (responseClass == null)
+        {
+            return null;
+        }
+        
+        // Special handling for ApiResponse
+        if (com.authlete.common.dto.ApiResponse.class.isAssignableFrom(responseClass))
+        {
+            com.authlete.common.dto.ApiResponse response = new com.authlete.common.dto.ApiResponse();
+            response.setResultCode("A" + statusCode + "000"); // Generic error code based on HTTP status
+            response.setResultMessage("HTTP " + statusCode + " response with empty/invalid body");
```

Comment:
I think it would be better to avoid introducing hypothetical result codes and messages, as they might give users the impression that these values were returned by the actual API or, in the worst case, conflict with a real result code. Would it be possible to remove these two lines?